src/xopen/__init__.py

```diff
@@ -412,7 +391,7 @@ def close(self) -> None:
     def __iter__(self):
         return self
 
-    def __next__(self) -> Union[str, bytes]:
+    def __next__(self) -> Union[str, bytes]:  # type: ignore # incompatible with type in IOBase
```

Comment:
This is something I ran into during my refactoring in #138 . The externally opened classes are slightly different in that they both support `t` and `b` modes. It would be much more in line with the rest of python if they are binary only, and a textwrapper is used around the pipedcompressionreader/writer objects. (Rather than interning a TextIOWrapper in the object as is done now). 

But that is something for later. I will make an issue about it.

Reply:
`TextIOWrapper` derives from `TextIOBase`, which derives from `IOBase`, so the problem that the return type changes in a derived class is still there. typeshed "solved" the issue the same way:
https://github.com/python/typeshed/blob/770724013de34af6f75fa444cdbb76d187b41875/stdlib/io.pyi#L145
I read a comment somewhere on the typeshed issue tracker stating that Python’s IO classes are just weird, and I agree.

It’s true we won’t have to deal with this when we make the Piped... classes work with binary data only. I agree and like the idea.

Reply:
Good to know that the issue is not something that is necessarily wrong on xopen's end. Thank you for the explanation.